blockchain/types/src/block.rs

```diff
@@ -123,14 +136,24 @@ impl Block {
     ///   block*
     /// * `root_element` - The root element for membership proofs
     /// * `block_contents` - Contents of the block.
+    /// * `timestamp` - The timestamp of the block in ms since Unix epoch. Only
+    ///    supported for BlockVersion 4 and higher.
     pub fn new(
         version: BlockVersion,
         parent_id: &BlockID,
         index: BlockIndex,
         cumulative_txo_count: u64,
         root_element: &TxOutMembershipElement,
         block_contents: &BlockContents,
+        timestamp: u64,
     ) -> Self {
+
+        let timestamp = if version.timestamps_are_supported() {
```

Comment:
If desired we could panic here if someone provides a timestamp for block version 3 or lower, throwing it away is kind of nice ergonomically, but is also a bit hidden

Reply:
I think panicing in the form of an assert seems reasonable, it's not something we expect to see ever happen and we be good to catch it if it does.